hw_resources_provider_node.py

```diff
@@ -251,21 +255,21 @@ def task_callback(ml_model, app_requirements, hw_constraints, node_status, hw):
         except Exception:
             model_path = ""
 
-    # Use model path if available
-    if model_path and model_path != "Error":
-        print(f"Using ONNX model path: {model_path}")
+    # Use RPTU hw predictor for their devices
+    if hw_selected in ["Zynq UltraScale+ ZCU102", "Zynq UltraScale+ ZCU104", "Ultra96-V2", "TySOM-3A-ZU19EG"]:
+        print(f"Using ONNX model path")
         try:
-            onnx_model = ONNXModel(model_path)
-            my_tensor = torch.rand(1,3,640,640, dtype=torch.float32)
-            upmem_layers.profiler_start(layer_mapping)
-            onnx_model.forward(my_tensor)
-            upmem_layers.profiler_end()
+            # Use RPTU
+            results = rptu_integration.onnx_ml_resource_estimation(os.path.dirname(__file__)+'/rptu_framework/model.onnx', hw_selected) # TODO: hw_selected should affect predictor
```

Comment:
NIT:
Please move the `'/rptu_framework/model.onnx'` to a variable upwards instead of hardcoding it